openvdb/points/PointSample.h

```diff
@@ -0,0 +1,359 @@
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+// IN NO EVENT SHALL THE COPYRIGHT HOLDERS' AND CONTRIBUTORS' AGGREGATE
+// LIABILITY FOR ALL CLAIMS REGARDLESS OF THEIR BASIS EXCEED US$250.00.
+//
+///////////////////////////////////////////////////////////////////////////
+
+/// @author Nick Avramoussis, Francisco Gochez
+///
+/// @file points/PointSample.h
+///
+/// @brief Sample a VDB Grid onto a VDB Points attribute
+
+#ifndef OPENVDB_POINTS_POINT_SAMPLE_HAS_BEEN_INCLUDED
+#define OPENVDB_POINTS_POINT_SAMPLE_HAS_BEEN_INCLUDED
+
+#include <type_traits> // enable_if
+
+#include "PointDataGrid.h"
+#include "PointGroup.h"
+
+
+namespace openvdb {
+OPENVDB_USE_VERSION_NAMESPACE
+namespace OPENVDB_VERSION_NAME {
+namespace points {
+
+/// @brief Samples values from an OpenVDB grid onto VDB Points
+///
+/// @details Performs sampling of an OpenVDB grid onto a collection of points and stores
+///          the results into a target attribute. If the attribute does not yet exist on the
+///          points, it will be added.  If it already exists it can be of a different type from the
+///          grid provided it can be cast with default constructors.
+///          The "Order" template parameter determines the form of sampling  which is performed.
+///          0 will use closest point, 1 will use tri-linear interpolation, and 2 will use
+///          tri-quadratic (see also the Sampler struct in tools/Interpolation.h) interpolation.
+/// @note    Samplers of order higher than 2 are not supported.
+///
+///  @param points           the PointDataGrid whose points will be sampled on to
+///  @param sourceGrid       VDB grid which will be sampled
+///  @param targetAttribute  a target attribute on the points which will hold samples. This
+///                          attribute will be created with the source grid type if it does
+///                          not exist
+///  @param includeGroups    names of groups to include
+///  @param excludeGroups    names of groups to exclude
+template <typename SourceGridT, typename PointDataGridT, size_t Order = 1>
+inline void sampleGrid(PointDataGridT& points,
+                       const SourceGridT& sourceGrid,
+                       const Name& targetAttribute,
```

Comment:
Could targetAttribute take a default argument of an empty string? That way, if the name was empty, the attribute name would simply be the name of the source grid?